server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -87,11 +102,24 @@ func (w *Worker) Work(ctx workflow.Context) {
 			}
 		}
 
-		// TODO: Validate revision before executing the workflow
+		var compareCommitResp activities.CompareCommitResponse
+		err = workflow.ExecuteActivity(ctx, w.Activities.CompareCommit, activities.CompareCommitRequest{
+			DeployRequestRevision:  msg.Revision,
+			LatestDeployedRevision: w.LatestDeployment.Revision,
+			Repo:                   msg.Repo,
+		}).Get(ctx, &compareCommitResp)
+		if err != nil {
+			logger.Error(ctx, fmt.Sprintf("Unable to compare deploy request commit with the latest deployed commit: %s", err.Error()))
+		}
+
+		if !w.isValidRevision(ctx, msg, compareCommitResp) {
+			continue
+		}
 
```

Comment:
Can this whole thing including fetching the latest deployment move into a function, that's sort of what i meant before in the previous comment.  I didn't mean just move the switch statement.